### PR TITLE
Fix null pointer in FoldInsertWithConstants.

### PR DESCRIPTION
**Pull request by LukasBanana**
Struct types are not supported in constant folding yet.

### CODE REVIEW

**Review by s-perron — COMMENTED** [review]
@LukasBanana Thanks for the fix, but could you add a test for this?

**Comment by LukasBanana** [comment]
I was trying to reproduce the issue with a small test, but so far it only occurred with one of our large Niagara shaders and I can't add them to this repository.
I can point you to the null pointer, though:
https://github.com/KhronosGroup/SPIRV-Tools/blob/main/source/opt/constants.cpp#L412

**Comment by s-perron** [comment]
I don't think it should be all that hard to reproduce in a folding test.

You will want to add a test to https://github.com/KhronosGroup/SPIRV-Tools/blob/0994ca45b663296d733e89c5d89dbd4347e065c1/test/opt/fold_test.cpp#L7088

The new tests will be similar to https://github.com/KhronosGroup/SPIRV-Tools/blob/0994ca45b663296d733e89c5d89dbd4347e065c1/test/opt/fold_test.cpp#L7297.

You will want to have an insert that is something like:

```
%4 = OpCompositeInsert  %struct_v2int_int_int %int_1 %struct_v2int_int_int_null 2
```

I believe that should reproduce your problem.

File: test/opt/fold_test.cpp [hunk 1]
```diff
@@ -7364,6 +7364,15 @@ ::testing::Values(
             "OpReturn\n" +
             "OpFunctionEnd",
         5, true)
+    // Test case 16: Don't fold when type cannot be deduced to a constant.
```
Comment (s-perron) [inline test/opt/fold_test.cpp:7367]: You need to add a comma in the line before this.

File: test/opt/fold_test.cpp [hunk 2]
```diff
@@ -7364,6 +7364,15 @@ ::testing::Values(
...
+    InstructionFoldingCase<bool>(
+        Header() +
+            "%main = OpFunction %void None %void_func\n" +
+            "%main_lab = OpLabel\n" +
+            "%4 = OpCompositeInsert %struct_v2int_int_int %int_1 %struct_v2int_int_int_null 2\n" +
+            "OpReturn\n" +
+            "OpFunctionEnd",
+        4, false),
```
Comment (s-perron) [inline test/opt/fold_test.cpp:7375]: You should double check, but I think you need to remove this comma.

**Review by s-perron — CHANGES_REQUESTED** [review]

**Comment by LukasBanana** [comment]
Thanks for your feedback. I don't have a proper workflow for the test cases yet, but will do that for the next PR.

**Pull request merged**